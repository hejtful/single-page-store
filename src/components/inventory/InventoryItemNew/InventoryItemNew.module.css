.wrapper {
  border: 1px solid #dbdbdb;
  margin-top: 1em;
  padding: 0.5em;
}

.formGroupInline {
  display: flex;
  justify-content: space-between;
}

.formGroupInline .formElement:first-child {
  margin-right: 0.5em;
}

.button {
  width: 100%;
}
