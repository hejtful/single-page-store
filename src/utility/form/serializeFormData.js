export function serializeFormData(form) {
  const formData = new FormData(form);
  const formObject = {};

  for (const key of formData.keys()) {
    formObject[key] = formData.get(key);
  }

  return formObject;
}
