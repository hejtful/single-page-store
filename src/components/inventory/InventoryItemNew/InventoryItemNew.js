import React from 'react';

import { serializeFormData } from 'utility/form/serializeFormData.js';

import styles from './InventoryItemNew.module.css';

export const InventoryItemNew = ({ onSubmit }) => {
  function handleSubmit(event) {
    event.preventDefault();

    onSubmit(serializeFormData(event.target));

    event.target.reset();
  }

  return (
    <div className={styles.wrapper}>
      <form onSubmit={handleSubmit}>
        <div className={styles.formGroupInline}>
          <input
            type="text"
            required
            name="title"
            placeholder="Title"
            className={styles.formElement}
          />

          <input
            type="number"
            step="0.01"
            min="0.01"
            required
            name="price"
            placeholder="Price"
            className={styles.formElement}
          />
        </div>

        <input
          type="url"
          required
          name="image"
          placeholder="Image URL"
          className={styles.formElement}
        />

        <textarea
          required
          name="description"
          placeholder="Description"
          className={`${styles.formElement} ${styles.textarea}`}
        />

        <button type="submit" className={styles.button}>
          Add Product
        </button>
      </form>
    </div>
  );
};
