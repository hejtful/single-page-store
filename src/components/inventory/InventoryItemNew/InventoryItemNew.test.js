import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { InventoryItemNew } from './InventoryItemNew';
import { serializeFormData } from 'utility/form/serializeFormData';

jest.mock('utility/form/serializeFormData');

const methods = {
  onSubmit: jest.fn(),
};

it('renders with empty values', () => {
  const { container } = render(<InventoryItemNew {...methods} />);

  expect(container).toMatchSnapshot();
});

it('should call onSubmit method with form object when submit button is clicked', () => {
  const expectedFormObject = {
    title: 'Test title',
    price: '1.25',
    image: 'http://test.image.com',
    description: 'A bit shorter description',
  };

  serializeFormData.mockReturnValue(expectedFormObject);

  const { getByRole } = render(<InventoryItemNew {...methods} />);

  userEvent.click(getByRole('button'));

  expect(methods.onSubmit).toBeCalledWith(expectedFormObject);
});
