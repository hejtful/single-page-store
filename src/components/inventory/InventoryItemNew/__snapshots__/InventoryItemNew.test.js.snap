// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with empty values 1`] = `
<div>
  <div
    class="wrapper"
  >
    <form>
      <div
        class="formGroupInline"
      >
        <input
          class="formElement"
          name="title"
          placeholder="Title"
          required=""
          type="text"
        />
        <input
          class="formElement"
          min="0.01"
          name="price"
          placeholder="Price"
          required=""
          step="0.01"
          type="number"
        />
      </div>
      <input
        class="formElement"
        name="image"
        placeholder="Image URL"
        required=""
        type="url"
      />
      <textarea
        class="formElement textarea"
        name="description"
        placeholder="Description"
        required=""
      />
      <button
        class="button"
        type="submit"
      >
        Add Product
      </button>
    </form>
  </div>
</div>
`;
